File: src/main/java/gg/mineral/practice/commands/settings/ToggleScoreboardCommand.java
package gg.mineral.practice.commands.settings;

import gg.mineral.practice.commands.PlayerCommand;

import gg.mineral.practice.managers.ProfileManager;
import gg.mineral.practice.util.messages.impl.ChatMessages;
import lombok.val;

public class ToggleScoreboardCommand extends PlayerCommand {

    public ToggleScoreboardCommand() {
        super("togglescoreboard");
    }

    @Override
    public void execute(org.bukkit.entity.Player pl, String[] args) {
        val profile = ProfileManager.getOrCreateProfile(pl);
        profile.setScoreboardEnabled(!profile.isScoreboardEnabled());
        ChatMessages.SCOREBOARD_TOGGLED.clone()
                .replace("%toggled%", profile.isScoreboardEnabled() ? "enabled" : "disabled")
                .send(pl);
    }
}

File: src/main/kotlin/gg/mineral/practice/commands/settings/ToggleScoreboardCommand.kt
package gg.mineral.practice.commands.settings

import dev.rollczi.litecommands.annotations.command.Command
import dev.rollczi.litecommands.annotations.context.Context
import dev.rollczi.litecommands.annotations.execute.Execute
import gg.mineral.practice.entity.Profile
import gg.mineral.practice.util.messages.impl.ChatMessages

@Command(name = "togglescoreboard")
class ToggleScoreboardCommand {
    @Execute
    fun execute(@Context profile: Profile) {
        profile.scoreboardEnabled = !profile.scoreboardEnabled
        profile.message(
            ChatMessages.SCOREBOARD_TOGGLED.clone()
                .replace("%toggled%", if (profile.scoreboardEnabled) "enabled" else "disabled")
        )
    }
}
